jinahub/encoder/image/clip.py

```diff
@@ -0,0 +1,80 @@
+from itertools import islice
+from typing import Optional, Iterable, Generator, Any
+
+import numpy as np
+import torch
+from jina import Executor, DocumentArray, requests
+
+import clip
+
+
+def _batch_generator(generator: Generator[int, Any, None], size: int):
+    while True:
```

Comment:
```suggestion
    for batch in islice(generator, size):
         yield batch
```

Shouldn't this work? I would as much as possible avoid `while` loops

Reply:
no, it won't work